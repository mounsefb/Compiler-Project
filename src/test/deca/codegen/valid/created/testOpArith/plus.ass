	TSTO #7
	BOV stack_overflow_error
	ADDSP #20
; start main program
; Main program
; Beginning of main instructions:
	LOAD #1, R2
	LOAD #2, R3
	LOAD #3, R4
	LOAD #4, R5
	LOAD #5, R6
	LOAD #6, R7
	ADD R7, R6
	ADD R6, R5
	ADD R5, R4
	ADD R4, R3
	ADD R3, R2
	STORE R2, 1(GB)
	LOAD #0x1.17ae14p4, R2
	LOAD #89, R3
	FLOAT R3, R3
	ADD R3, R2
	BOV overflow_error
	STORE R2, 2(GB)
	LOAD 1(GB), R1
	WINT
	WNL
	LOAD 2(GB), R1
	WFLOATX
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
; code des m√©thodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
