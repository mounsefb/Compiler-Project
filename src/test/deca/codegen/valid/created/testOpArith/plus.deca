// Description :
//    Test d'opérations de somme avec des variables de types différents
//
// Résultats :
//    21
//    0x1.a9eb84p+6



{
    int somme = 1 + (2 + (3 + (4 + (5 + 6))));
    float shesh = 17.48 + 89;
    println(somme);
    printx(shesh);
}


