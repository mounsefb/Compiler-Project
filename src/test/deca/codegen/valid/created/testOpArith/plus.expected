21
0x1.a9eb84p+6
